packages/db/src/db.ts

```diff
@@ -45,26 +53,62 @@ export class TruffleDB {
     return await execute(this.schema, document, null, this.context, variables);
   }
 
-  async loadCompilations(result: WorkflowCompileResult) {
-    const saga = generateCompileLoad(result, {
+  async getWorkspaceResponse(generatorRequest: WorkspaceRequest) {
+    const { request, variables }: WorkspaceRequest = generatorRequest;
+
+    const response: WorkspaceResponse = await this.query(request, variables);
+
+    return response;
+  }
+  
+  async loadNames(
+    project: DataModel.IProject,
+    contractsByCompilation: Array<DataModel.IContract[]>
+  ) {
+    const namesLoader = generateNamesLoad(
+      toIdObject(project),
+      contractsByCompilation
+    );
+    let curNames = namesLoader.next();
+
+    while (!curNames.done) {
+      const namesResponse = await this.getWorkspaceResponse(curNames.value);
+      curNames = namesLoader.next(namesResponse);
+    }
+  }
+
+  async loadProject(): Promise<DataModel.IProject> {
+    const projectRequest = projectLoadGenerate({
       directory: this.context.workingDirectory
-    });
+    }).next();
+
+    const response = await this.getWorkspaceResponse(projectRequest.value);
+    const projectResponse = response.data.workspace.projectsAdd.projects[0];
+
+    return projectResponse;
+  }
+
+  async loadCompilations(
+    project: DataModel.IProject,
+    result: WorkflowCompileResult,
+    names: boolean
```

Comment:
I wondering if this should go into an options object. Unnamed random booleans are always worrisome to my on the caller side.